src/common/errors.js

```diff
@@ -1,7 +1,18 @@
 'use strict';
 const util = require('util');
 
-class RippleError extends Error {
+// this is needed because extending builtins doesn't work in babel 6.x
+function extendableBuiltin(cls) {
+  function ExtendableBuiltin() {
+    cls.apply(this, arguments);
+  }
+  ExtendableBuiltin.prototype = Object.create(cls.prototype);
+  Object.setPrototypeOf(ExtendableBuiltin, cls);
+
+  return ExtendableBuiltin;
+}
+
+class RippleError extends extendableBuiltin(Error) {
```

Comment:
Ok. So got everything working, and as @darkdarkdragon it doesn't work as I wrote it originally. However, this works:

```
'use strict'

class MyError {
  constructor (message) {

    // Set this.message
    Object.defineProperty(this, 'message', {
      configurable: true,
      enumerable: false,
      value: message !== undefined ? String(message) : ''
    })

    // Set this.name
    Object.defineProperty(this, 'name', {
      configurable: true,
      enumerable: false,
      value: this.constructor.name
    })

    // Set this.stack
    Error.captureStackTrace(this, this.constructor)
  }
}

require('util').inherits(MyError, Error)



console.log((new MyError('foo') instanceof Error))
console.log((new MyError('foo') instanceof MyError))
```

```
true
true
```